### PR TITLE
[docs] Add example of starting up a registry on VIC [skip ci]

### PR DESCRIPTION
**Pull request by sflxn**
Provide an example of how to start up a non-authenticating
Docker registry using VIC.

### CODE REVIEW

File: doc/application_workflow/private_registry.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Running a private registry with VIC
+
+In this example, we will run a private Docker registry on VIC and push and pull images using VIC.  VMWare also offers an enterprise ready registry named [Harbor](https://github.com/vmware/harbor) that can be used in place of the base Docker registry.
```
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: `VMware`

File: doc/application_workflow/private_registry.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## Prerequisite
+
+Before going through this example, we need to re-emphasize some concepts around Docker and VIC.  The following examples are shown using Linux.  For Windows and Mac users, these examples should not differ much.
+
+Installing VIC will also require installing Docker locally.  When Docker is installed, we will get both a client (CLI or command line interface) and a daemon that handles all local container operations.  Local containers are those that run on the user local machine instead of a VMWare vSphere/ESXi environment.  The CLI is important as it will be most user's touchpoint for working with containers on VIC and on their local system.  The distinction between using the CLI against both environment is very important in this example.  By default, the CLI will use the local Docker daemon.  After setting some environment variables, the CLI can be instructed to send all operations to VIC instead of the local Docker daemon.  The two environment variable are DOCKER_HOST and DOCKER_API_VERSION.
```
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: `both environments`

File: doc/application_workflow/private_registry.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## Workflow
+
+In terminal #1: (local Docker)
+
+1. Open a terminal and make sure that it will use the local Docker daemon.  At a command prompt, issue  
+```
+    $> unset DOCKER_HOST
+```
+2. Install a VCH for a private registry using vic-machine
```
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: This shows up as `1.` in the rendered file.

File: doc/application_workflow/private_registry.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+3. Run Docker's registry on the first VCH
```
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: A link to the docs for Docker's registry will be helpful. Perhaps https://docs.docker.com/registry/deploying?

File: doc/application_workflow/private_registry.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+4. Install a second VCH for running applications.  Make sure to specify --docker-insecure-registry.  This will make sure this second VIC can pull images from the insecure registry in the first VCH.
```
Comment (jzt) [inline doc/application_workflow/private_registry.md]: Maybe reword this to something like:

"Install a second VCH for running applications, making sure to specify `--docker-insecure-registry` to ensure that this second VIC can pull images from the insecure registry in the first VCH"

That way it won't look that weird without a period at the end, and you can remove it and the one on the next line (5) for consistency. :)

Comment (jzt) [inline doc/application_workflow/private_registry.md]: Although mine sounds somewhat like a run-on.  Just my $0.02!

File: doc/application_workflow/private_registry.md (hunk 1 above)
Comment (matthewavery) [inline doc/application_workflow/private_registry.md]: I always capitalize the 'W' lol

File: doc/application_workflow/private_registry.md (hunk 5 above)
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: * I believe the flag is `--insecure-registry` now, based on my local branch that's synced with master.
* Formatting - specify --docker-insecure-registry -> specify \`---insecure-registry\`

File: doc/application_workflow/private_registry.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+```
+[Service]
+Type=notify
+# the default is not to use systemd for cgroups because the delegate issues still
+# exists and systemd currently does not support the cgroup feature set required
+# for containers run by docker
+ExecStart=/usr/bin/dockerd --tls=false -H fd:// --insecure-registry 192.168.218.138:5000
+```
+
+Also, note in terminal 1 above, in the second step where we run the Docker registry.  In that command, we specify the necessary environment variables before the docker run command.  On our Linux machine, this sets the variables for the duration of the operation, and once the docker run finishes, those variables are reverted back to their previous values.  Also, note that we run registry:2.  It is important not to specify registry:2.0 in this example.  Registry 2.0 had issues that prevents the example above from working.
```
Comment (jzt) [inline doc/application_workflow/private_registry.md]: s/Registry 2.0 had issues/Registry 2.0 has issues

**Comment by jzt** [comment]
LGTM with minor suggestions

File: doc/application_workflow/private_registry.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+3. Run Docker's registry on the first VCH
+4. Install a second VCH for running applications.  Make sure to specify --docker-insecure-registry.  This will make sure this second VIC can pull images from the insecure registry in the first VCH.
+5. At a terminal command prompt, using regular Docker, tag the images to be destined for the registry.
+6. Modify the docker systemd config file to allow pushing to an insecure registry
+7. Restart the docker daemon
+8. Push the image using the full tagged name (including host IP and port)
+
+In terminal #2: (VIC VCH)
+
+1. Open a terminal and make sure it is using the second VCH.  At a command promt, issue
```
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: `prompt`

Comment (jzt) [inline doc/application_workflow/private_registry.md]: good eye :)

File: doc/application_workflow/private_registry.md (hunk 6 above)
Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: `reverted back` -> `reverted`

Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: Please reword this to something like

`In the second step for terminal 1 above, we specify...`

File: doc/application_workflow/private_registry.md (hunk 1 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: done

File: doc/application_workflow/private_registry.md (hunk 2 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: sounds better

File: doc/application_workflow/private_registry.md (hunk 3 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: fixed

File: doc/application_workflow/private_registry.md (hunk 4 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: now linked to just basic registry url.

File: doc/application_workflow/private_registry.md (hunk 5 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: fixed

File: doc/application_workflow/private_registry.md (hunk 6 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: done

File: doc/application_workflow/private_registry.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+INFO[2016-10-08T17:31:06-07:00]   DOCKER_HOST=192.168.218.138:2375           
+INFO[2016-10-08T17:31:06-07:00]                                              
+INFO[2016-10-08T17:31:06-07:00]                                              
+INFO[2016-10-08T17:31:06-07:00] Connect to docker:                           
+INFO[2016-10-08T17:31:06-07:00] docker -H 192.168.218.138:2375 info          
+INFO[2016-10-08T17:31:06-07:00] Installer completed successfully             
+
+$> DOCKER_HOST=tcp://192.168.218.138:2375 DOCKER_API_VERSION=1.23 docker run -d -p 5000:5000 --name registry registry:2
+
+$> ./vic-machine-linux create --target=192.168.218.207 --image-store=datastore1 --name=vic-app --user=root --password=vagrant --compute-resource="/ha-datacenter/host/esxbox.localdomain/Resources" --bridge-network=vic-network --no-tls --volume-store=datastore1/vic-app:default --force --docker-insecure-registry 192.168.218.138
```
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: fixed

Comment (anchal-agrawal) [inline doc/application_workflow/private_registry.md]: Looks like it: https://github.com/vmware/vic/blob/master/cmd/vic-machine/create/create.go#L383

File: doc/application_workflow/private_registry.md (hunk 7 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: fixed.

File: doc/application_workflow/private_registry.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+8. Push the image using the full tagged name (including host IP and port)
+
+In terminal #2: (VIC VCH)
+
+1. Open a terminal and make sure it is using the second VCH.  At a command promt, issue  
+```
+    $> export DOCKER_HOST=tcp://<VCH_IP>:<VCH_PORT>
+    $> export DOCKER_API_VERSION=1.23
+```
+2. Pull the image from the registry VCH
```
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: should be fixed.

File: doc/application_workflow/private_registry.md (hunk 6 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: done.

Comment (sflxn) [inline doc/application_workflow/private_registry.md]: done

**Comment by anchal-agrawal** [comment]
LGTM

**Comment by sflxn** [comment]
ping

File: doc/application_workflow/private_registry.md (hunk 6 above)
Comment (sflxn) [inline doc/application_workflow/private_registry.md]: done

File: doc/application_workflow/private_registry.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+INFO[2016-10-08T17:31:06-07:00]   DOCKER_HOST=192.168.218.138:2375           
+INFO[2016-10-08T17:31:06-07:00]                                              
+INFO[2016-10-08T17:31:06-07:00]                                              
+INFO[2016-10-08T17:31:06-07:00] Connect to docker:                           
+INFO[2016-10-08T17:31:06-07:00] docker -H 192.168.218.138:2375 info          
+INFO[2016-10-08T17:31:06-07:00] Installer completed successfully             
+
+$> DOCKER_HOST=tcp://192.168.218.138:2375 DOCKER_API_VERSION=1.23 docker run -d -p 5000:5000 --name registry registry:2
+
+$> ./vic-machine-linux create --target=192.168.218.207 --image-store=datastore1 --name=vic-app --user=root --password=vagrant --compute-resource="/ha-datacenter/host/esxbox.localdomain/Resources" --bridge-network=vic-network --no-tls --volume-store=datastore1/vic-app:default --force --insecure-registry 192.168.218.138
```
Comment (hickeng) [inline doc/application_workflow/private_registry.md]: This should **not** be using the same bridge network as the other VCH. That will lead to IP conflicts.

File: doc/application_workflow/private_registry.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        $> export DOCKER_API_VERSION=1.23
+    ```
+2. Pull the image from the registry VCH
+
+### Example run
+
+terminal 1:
+```
+$> unset DOCKER_HOST
+$> ./vic-machine-linux create --target=192.168.218.207 --image-store=datastore1 --name=vic-registry --user=root --password=vagrant --compute-resource="/ha-datacenter/host/esxbox.localdomain/Resources" --bridge-network=vic-network --no-tls --volume-store=datastore1/registry:default --force
```
Comment (hickeng) [inline doc/application_workflow/private_registry.md]: Suggest you update this to use --client-network-ip=fqdn, or to allow the registry container to be attached directly to a vsphere network with a static IP.
Either of those approaches will provide a stable registry endpoint for people to address.
